marketplacetests/marketplace/app.py

```diff
@@ -82,34 +76,32 @@ def wait_for_notification_message_displayed(self, message=None):
     def wait_for_notification_message_not_displayed(self):
         self.wait_for_element_not_displayed(*self._notification_locator)
 
+    def wait_for_page_loaded(self):
+        Wait(self.marionette).until(
+            expected.element_present(*self._body_loaded_locator))
+        Wait(self.marionette).until(
+            expected.element_present(*self._page_loaded_locator))
+
     @property
     def notification_message(self):
         return self.marionette.find_element(*self._notification_locator).text
 
-    @property
-    def popular_apps(self):
-        self.show_popular_apps()
-        from marketplacetests.marketplace.regions.search_results import Result
-        return [Result(self.marionette, app) for app in self.marionette.find_elements(*self._gallery_apps_locator)]
-
-    def search(self, term):
-        self.wait_for_element_displayed(*self._search_locator)
-        search_box = self.marionette.find_element(*self._search_locator)
-
-        # search for the app
+    def _perform_search(self, term):
+        self.wait_for_page_loaded()
```

Comment:
It would seem to make more sense to perform the wait after an interaction that causes a page load, rather than before an interaction that requires the page to be loaded.